src/CodeExplorerBundle/Twig/SourceCodeExtension.php

```diff
@@ -41,13 +41,16 @@ public function setController($controller)
     public function getFunctions()
     {
         return array(
-            new \Twig_SimpleFunction('show_source_code', array($this, 'showSourceCode'), array('is_safe' => array('html'), 'needs_environment' => true)),
+            new \Twig_SimpleFunction('show_source_code', array($this, 'showSourceCode'), array(
+                'is_safe' => array('html'),
+                'node_class' => 'CodeExplorerBundle\Twig\Node\TemplateAwareFunction'
+            )),
         );
     }
 
-    public function showSourceCode(\Twig_Environment $twig, \Twig_Template $template)
+    public function showSourceCode(\Twig_Template $template)
     {
-        return $twig->render('@CodeExplorer/source_code.html.twig', array(
+        return $template->getEnvironment()->render('@CodeExplorer/source_code.html.twig', array(
```

Comment:
@stof, using the `needs_environment` option isn't possible here, because it requires more complicated code. What about to inject the `Twig_Environment` instance to the extention?